appstudio-controller/controllers/appstudio.redhat.com/devsandbox_deployment_controller.go

```diff
@@ -0,0 +1,214 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package appstudioredhatcom
+
+import (
+	"context"
+	"fmt"
+
+	codereadytoolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	applicationv1alpha1 "github.com/redhat-appstudio/application-api/api/v1alpha1"
+	apierr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+const (
+	// deploymentTargetClaimLabel is the label indicating the DeploymentTargetClaim that's associated with the object
+	deploymentTargetClaimLabel = "appstudio.openshift.io/dtc"
+)
+
+// DevsandboxDeploymentReconciler reconciles a SpaceRequest object
+type DevsandboxDeploymentReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargetclaims/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargets,verbs=get;list;create;watch;update;patch;delete
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=deploymenttargets/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacerequests,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=toolchain.dev.openshift.com,resources=spacerequests/status,verbs=get;update;patch
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// Modify the Reconcile function to compare the state specified by
+// the DeploymentTargetClaim object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *DevsandboxDeploymentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx).WithValues("name", req.Name, "namespace", req.Namespace, "component", "devsandboxDeployment")
+
+	spacerequest := codereadytoolchainv1alpha1.SpaceRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      req.Name,
+			Namespace: req.Namespace,
+		},
+	}
+
+	if err := r.Client.Get(ctx, client.ObjectKeyFromObject(&spacerequest), &spacerequest); err != nil {
+		// Don't requeue if the requested object is not found/deleted.
+		if apierr.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
```

Comment:
Can you add a quick sanity test here, checking that the `SpaceRequest` is in ready state? If the controller isn't ready, we can just `return ctrl.Result{}, nil`. I know we already have a predicate for this on the controller, but it can't hurt to verify here as well.

Reply:
Added. Thanks